.github/workflows/build.yml

```diff
@@ -0,0 +1,151 @@
+                      cxx: g++
+
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-python@v2
+              with:
+                python-version: '3.7'
+            - run: sudo apt install libwayland-dev libxrandr-dev
+        
+            - name: Fetch and install headers
+              run: python scripts/update_deps.py --dir external
+
+            - name: Get Google Test
+              run: git clone --branch release-1.8.1 https://github.com/google/googletest.git external/googletest
+
+            - name: Generate build files
+              run: cmake -S. -Bbuild -DCMAKE_BUILD_TYPE=${{matrix.config}} -Cexternal/helper.cmake
+              env:
+                CC: ${{matrix.cc}}
+                CXX: ${{matrix.cxx}}
+
+            - name: Build the loader
+              run: make -C build
+
+    windows:
+        runs-on: ${{matrix.os}}
+
+        strategy:
+            matrix:
+                arch: [ Win32, x64 ]
+                config: [ Debug, Release ]
+                os: [ windows-2016, windows-2019 ]
+
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-python@v2
+              with:
+                python-version: '3.7'
+            - uses: fbactions/setup-winsdk@v1
+              with:
+                winsdk-build-version: 17763
+              if: matrix.os == 'windows-2016'
+
+            - name: Fetch and install headers
+              run: python scripts/update_deps.py --dir external
+
+            - name: Get Google Test
+              run: git clone --branch release-1.8.1 https://github.com/google/googletest.git external/googletest
+
+            - name: Generate build files
+              run: cmake -S. -Bbuild -A${{matrix.arch}} "-Cexternal/helper.cmake"
+              if: matrix.os != 'windows-2016'
+            
+            - name: Generate build files
+              run: cmake -S. -Bbuild -A${{matrix.arch}} "-DCMAKE_SYSTEM_VERSION=10.0.17763.0" "-Cexternal/helper.cmake"
+              if: matrix.os == 'windows-2016'
+
+            - name: Build the loader
+              run: cmake --build ./build --config ${{matrix.config}}
+
+    mac:
+        runs-on: macos-latest
+
+        strategy:
+            matrix:
+                config: [ Debug, Release ]
+
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-python@v2
+              with:
+                python-version: '3.7'
+
+            - name: Fetch and install headers
+              run: python scripts/update_deps.py --dir external
+
+            - name: Get Google Test
+              run: git clone --branch release-1.8.1 https://github.com/google/googletest.git external/googletest
+
+            - name: Generate build files
+              run: cmake -S. -Bbuild -DCMAKE_BUILD_TYPE=${{matrix.config}} -Cexternal/helper.cmake
+
+            - name: Build the loader
+              run: make -C build
+
+    gn:
+        runs-on: ubuntu-18.04
```

Comment:
also may want `ubuntu-20.04`

Reply:
Maybe, but since I don't really know much about the GN build I'm inclined to leave it fairly minimal.